content/guides/javascript/get-program-accounts.md

```diff
@@ -0,0 +1,307 @@
+  - `lamports`: `number`, number of lamports assigned to the account
+  - `owner`: `string`, The base58 encoded pubkey of the program the account has
+    been assigned to
+  - `data`: `string` | `object` - data associated with the account, either as
+    encoded binary data or JSON format depending on the provided encoding
+    parameter
+  - `executable`: `boolean`, Indication if the account contains a program
+  - `rentEpoch`: `number`, The epoch at which this account will next owe rent
+
+## Deep Dive
+
+`getProgramAccounts` is a versatile RPC method that returns all accounts owned
+by a program. We can use `getProgramAccounts` for a number of useful queries,
+such as finding:
+
+- All token accounts for a particular wallet
+- All token accounts for a particular mint (i.e. All
+  [SRM](https://www.projectserum.com/) holders)
+- All custom accounts for a particular program (i.e. All
+  [Mango](https://mango.markets/) users)
+
+Despite its usefulness, `getProgramAccounts` is often misunderstood due to its
+current constraints. Many of the queries supported by `getProgramAccounts`
+require RPC nodes to scan large sets of data. These scans are both memory and
+resource intensive. As a result, calls that are too frequent or too large in
+scope can result in connection timeouts. Furthermore, at the time of this
+writing, the `getProgramAccounts` endpoint does not support pagination. If the
+results of a query are too large, the response will be truncated.
+
+To get around these current constraints, `getProgramAccounts` offers a number of
+useful parameters: namely, `dataSlice` and the `filters` options `memcmp` and
+`dataSize`. By providing combinations of these parameters, we can reduce the
+scope of our queries down to manageable and predictable sizes.
+
+A common example of `getProgramAccounts` involves interacting with the
+[SPL-Token Program](https://spl.solana.com/token). Requesting all accounts owned
+by the Token Program with a
+[basic call](../references/accounts.md#get-program-accounts) would involve an
+enormous amount of data. By providing parameters, however, we can efficiently
+request just the data we intend to use.
+
+### `filters`
+
+The most common parameter to use with `getProgramAccounts` is the `filters`
+array. This array accepts two types of filters,`dataSize` and `memcmp`. Before
+using either of these filters, we should be familiar with how the data we are
+requesting is laid out and serialized.
+
+#### `dataSize`
+
+In the case of the Token Program, we can see that
+[token accounts are 165 bytes in length](https://github.com/solana-labs/solana-program-library/blob/08d9999f997a8bf38719679be9d572f119d0d960/token/program/src/state.rs#L86-L106).
+Specifically, a token account has eight different fields, with each field
+requiring a predictable number of bytes. We can visualize how this data is laid
+out using the below illustration.
+
+![Account Size](./get-program-accounts/account-size.png)
```

Comment:
Fix this URL to be the full path too. Also check the other PRs in case I've missed these! Search for `(./` with regex turned off.